pkg/resource/logging-credentials/reconciler.go

```diff
@@ -0,0 +1,76 @@
+package loggingcredentials
+
+import (
+	"context"
+	"fmt"
+
+	loggedcluster "github.com/giantswarm/logging-operator/pkg/logged-cluster"
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	apimachineryerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// Reconciler implements a reconciler.Interface to handle
+// Logging Credentials: store and maintain logging credentials
+type Reconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+// ReconcileCreate ensures a secret exists for the given cluster.
+func (r *Reconciler) ReconcileCreate(ctx context.Context, lc loggedcluster.Interface) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+
+	logger.Info(fmt.Sprintf("loggingcredentials checking secret %s/%s", LoggingCredentialsSecretMeta(lc).Namespace, LoggingCredentialsSecretMeta(lc).Name))
+
+	var loggingCredentialsSecret v1.Secret
+
+	// Check if secrets exist / retrieve existing secret
+	err := r.Client.Get(ctx, types.NamespacedName{Name: LoggingCredentialsSecretMeta(lc).Name, Namespace: LoggingCredentialsSecretMeta(lc).Namespace}, &loggingCredentialsSecret)
+	if err != nil {
+		if apimachineryerrors.IsNotFound(err) {
+			logger.Info("loggingcredentials secret not found, initializing one")
+			// Create basic secret
+			loggingCredentialsSecret, err = GenerateLoggingCredentialsBasicSecret(lc)
+		}
+		if err != nil {
+			return ctrl.Result{}, errors.WithStack(err)
+		}
```

Comment:
This is a strange flow. I don't know if it works by chance or deliberately but I think it's hard to understand at first glance that the `err` variable is getting overwritten by whatever error is returned by `GenerateLoggingCredentialsBasicSecret`. So the `if err != nil` from below that, it will check that error, instead of the one coming from fetching the secret from the k8s api. I would remove the error from `GenerateLoggingCredentialsBasicSecret`.

Reply:
This code was removed / changed with commit 81c1d49d6228b4aed1d0f5028ebe91129e6ed267 :heavy_check_mark: